graphium/finetuning/finetuning.py

```diff
@@ -0,0 +1,56 @@
+from typing import Iterable, List, Dict, Tuple, Union, Callable, Any, Optional, Type
+
+import pytorch_lightning as pl
+
+from torch.optim.optimizer import Optimizer
+from pytorch_lightning.callbacks import BaseFinetuning
+
+
+class MolecularFinetuning(BaseFinetuning):
```

Comment:
Nothing from the class suggests that it's for Molecules only. I would cal it something more generic, like `GraphFinetuning` or `FullGraphNetworkFinetuning` (if specific to the `FullGraphNetwork` class, which it seems to be since it's using `model.task_heads` somewhere)